### PR TITLE
fix: [ #107 ] Delete images

### PR DESCRIPTION
**Pull request by CloudyPadmal**
Fixes issue #107

Changes:
- Implemented single image deletion
- Implemented multiple image deletion

Screenshots for the change: 
- Multiple image deletion
Multiple Image Deletion

- Single image deletion
Single Image

- Delete confirmation
Delete confirmation

### CODE REVIEW

**Comment by Abhi2424shek** [comment]
Can you see the failing checks?

**Comment by CloudyPadmal** [comment]
Yes @Abhi2424shek !!
> vn.mbm.phimp.me.CameraActivityTest > cameraActivityTest[test(AVD) - 4.4.4] FAILED 
> vn.mbm.phimp.me.MainActivityTest > waitFor5Seconds[test(AVD) - 4.4.4] FAILED 
> vn.mbm.phimp.me.SettingsActivityTest > settingsActivityTest[test(AVD) - 4.4.4] FAILED 
> vn.mbm.phimp.me.UploadActivityTest > waitFor5Seconds[test(AVD) - 4.4.4] FAILED 

Similar error message :
> java.lang.RuntimeException: Waited for the root of the view hierarchy to have window focus and not be requesting layout for over 10 seconds. If you specified a non default root matcher, it may be picking a root that never takes focus. Otherwise, something is seriously wrong. Selected Root:

>	Root{application-window-token=android.view.ViewRootImpl$W@b2fd7ec8, window-token=android.view.ViewRootImpl$W@b2fd7ec8, has-window-focus=false, layout-params-type=1, layout-params-string=WM.LayoutParams{(0,0)(fillxfill) sim=#100 ty=1 fl=#1810100 pfl=0x8 wanim=0x10302a1}, decor-view-string=DecorView{id=-1, visibility=VISIBLE, width=480, height=800, has-focus=true, has-focusable=true, has-window-focus=false, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1}}

**Comment by dynamitechetan** [comment]
same error for #120

**Comment by Abhi2424shek** [comment]
Opened issue for it. High priority.

**Comment by Abhi2424shek** [comment]
Fixes it in #122 after it is merged restart the build.

**Comment by CloudyPadmal** [comment]
@Abhi2424shek can you review if this is okay? :)

**Comment by pa1pal** [comment]
@CloudyPadmal It requires long press to select all the images.
It should be long press for first image and then select on simple tap. 
What do you think?

**Comment by CloudyPadmal** [comment]
I was wondering! :) I'll implement single click and add a menu item to deselect the selected ones?

**Comment by CloudyPadmal** [comment]
@pa1pal is it okay now? :)

File: app/src/main/java/vn/mbm/phimp/me/CacheStore.java [hunk 1]
```diff
@@ -156,4 +156,47 @@ public void clearCache(){
...
+			deletableFile.delete();
+		}/*
+		String imageFiles[] = cacheDirectory.list();
+		// Search for the image and delete it
+		for (String imageFile : imageFiles) {
+			if (imageFile.contentEquals(cacheMap.get(path).toString())) {
+				File deletableFile = new File(
+						Environment.getExternalStorageDirectory() + cacheDir + imageFile);
+				if (deletableFile.exists()) {
+					deletableFile.delete();
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/CacheStore.java]: Delete the not needed commented out code

File: app/src/main/java/vn/mbm/phimp/me/ImageItem.java [hunk 2]
```diff
@@ -7,7 +7,17 @@
...
 
 class ImageItem {
+
     int id;
     Bitmap img;
     String path;
+    boolean isSelected;
+
+    public boolean isSelected() {
+        return isSelected;
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/ImageItem.java]: I think you can directory call the boolean with the function right?

File: app/src/main/java/vn/mbm/phimp/me/CacheStore.java (hunk 1 above)
Comment (CloudyPadmal) [inline app/src/main/java/vn/mbm/phimp/me/CacheStore.java]: Fixed it :D

File: app/src/main/java/vn/mbm/phimp/me/ImageItem.java (hunk 2 above)
Comment (CloudyPadmal) [inline app/src/main/java/vn/mbm/phimp/me/ImageItem.java]: I used getters and setters! I've changed it to a single variable now :D

**Comment by CloudyPadmal** [comment]
@Abhi2424shek how's this now? :)

**Comment by CloudyPadmal** [comment]
@pa1pal can you review this if you got time? :)

**Comment by CloudyPadmal** [comment]
@pa1pal  @Abhi2424shek Can you please take a look? :)

File: app/src/main/res/menu/menu_gallery.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        app:showAsAction="ifRoom" />
+
+    <item
+        android:id="@+id/menu_gallery_deselect_selected"
+        android:icon="@drawable/ic_deselect_white_24dp"
+        android:title="@string/menu_title_deselect"
+        android:orderInCategory="1"
+        app:showAsAction="ifRoom" />
+
+</menu>
```
Comment (Abhi2424shek) [inline app/src/main/res/menu/menu_gallery.xml]: Add a new line at EOF

File: app/src/main/res/layout/photoitem_local.xml [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+
+    <CheckBox
+        android:id="@+id/localPhotoSelector"
+        android:layout_height="wrap_content"
+        android:layout_width="wrap_content"
+        android:visibility="gone"
+        android:layout_alignParentEnd="true"
+        android:layout_alignParentRight="true"/>
+
+</RelativeLayout>
```
Comment (Abhi2424shek) [inline app/src/main/res/layout/photoitem_local.xml]: Add a new line at EOF

File: app/src/main/res/menu/menu_gallery.xml (hunk 3 above)
Comment (CloudyPadmal) [inline app/src/main/res/menu/menu_gallery.xml]: Added! :+1:

File: app/src/main/res/layout/photoitem_local.xml (hunk 4 above)
Comment (CloudyPadmal) [inline app/src/main/res/layout/photoitem_local.xml]: Added :+1:

**Comment by Abhi2424shek** [comment]
Merge conflicts are still there.

**Comment by CloudyPadmal** [comment]
@Abhi2424shek I was just solving them :relaxed: Now building!

**Comment by CloudyPadmal** [comment]
@pa1pal @Abhi2424shek since UI and other functionalities have changed after this PR, I've made a new PR with conflicts resolved! :) So I'm closing this one! A new PR is open :tada:

**Pull request closed without merging**